tests/checks/mock/test_haproxy.py

```diff
@@ -0,0 +1,138 @@
+import copy
+
+# 3p
+import mock
+
+# project
+from tests.checks.common import AgentCheckTest
+
+
+MOCK_DATA = """# pxname,svname,qcur,qmax,scur,smax,slim,stot,bin,bout,dreq,dresp,ereq,econ,eresp,wretr,wredis,status,weight,act,bck,chkfail,chkdown,lastchg,downtime,qlimit,pid,iid,sid,throttle,lbtot,tracked,type,rate,rate_lim,rate_max,check_status,check_code,check_duration,hrsp_1xx,hrsp_2xx,hrsp_3xx,hrsp_4xx,hrsp_5xx,hrsp_other,hanafail,req_rate,req_rate_max,req_tot,cli_abrt,srv_abrt,
+a,FRONTEND,,,1,2,12,1,11,11,0,0,0,,,,,OPEN,,,,,,,,,1,1,0,,,,0,1,0,2,,,,0,1,0,0,0,0,,1,1,1,,,
+a,BACKEND,0,0,0,0,12,0,11,11,0,0,,0,0,0,0,UP,0,0,0,,0,1221810,0,,1,1,0,,0,,1,0,,0,,,,0,0,0,0,0,0,,,,,0,0,
+b,FRONTEND,,,1,2,12,11,11,0,0,0,0,,,,,OPEN,,,,,,,,,1,2,0,,,,0,0,0,1,,,,,,,,,,,0,0,0,,,
+b,i-1,0,0,0,1,,1,1,0,,0,,0,0,0,0,UP,1,1,0,0,1,1,30,,1,3,1,,70,,2,0,,1,1,,0,,,,,,,0,,,,0,0,
+b,i-2,0,0,1,1,,1,1,0,,0,,0,0,0,0,UP,1,1,0,0,0,1,0,,1,3,2,,71,,2,0,,1,1,,0,,,,,,,0,,,,0,0,
+b,i-3,0,0,0,1,,1,1,0,,0,,0,0,0,0,UP,1,1,0,0,0,1,0,,1,3,3,,70,,2,0,,1,1,,0,,,,,,,0,,,,0,0,
+b,i-4,0,0,0,1,,1,1,0,,0,,0,0,0,0,DOWN,1,1,0,0,0,1,0,,1,3,3,,70,,2,0,,1,1,,0,,,,,,,0,,,,0,0,
+b,i-5,0,0,0,1,,1,1,0,,0,,0,0,0,0,MAINT,1,1,0,0,0,1,0,,1,3,3,,70,,2,0,,1,1,,0,,,,,,,0,,,,0,0,
+b,BACKEND,0,0,1,2,0,421,1,0,0,0,,0,0,0,0,UP,6,6,0,,0,1,0,,1,3,0,,421,,1,0,,1,,,,,,,,,,,,,,0,0,
+c,i-1,0,0,0,1,,1,1,0,,0,,0,0,0,0,UP,1,1,0,0,1,1,30,,1,3,1,,70,,2,0,,1,1,,0,,,,,,,0,,,,0,0,
+c,i-2,0,0,0,1,,1,1,0,,0,,0,0,0,0,DOWN,1,1,0,0,1,1,30,,1,3,1,,70,,2,0,,1,1,,0,,,,,,,0,,,,0,0,
+c,BACKEND,0,0,1,2,0,421,1,0,0,0,,0,0,0,0,UP,6,6,0,,0,1,0,,1,3,0,,421,,1,0,,1,,,,,,,,,,,,,,0,0,
+"""
+
+
+class MockResponse(object):
+    def __init__(self):
+        self.content = MOCK_DATA
+
+    def raise_for_status(self):
+        pass
+
+
+class TestCheckHAProxy(AgentCheckTest):
+    CHECK_NAME = 'haproxy'
+
+    BASE_CONFIG = {
+        'init_config': None,
+        'instances': [
+            {
+                'url': 'http://localhost/admin?stats',
+                'collect_status_metrics': True,
+            }
+        ]
+    }
+
+    def _assert_agg_statuses(self, count_status_by_service):
+        if count_status_by_service:
+            self.assertMetric('haproxy.count_per_status', value=1, tags=['status:available', 'service:a'])
+            self.assertMetric('haproxy.count_per_status', value=4, tags=['status:available', 'service:b'])
+            self.assertMetric('haproxy.count_per_status', value=2, tags=['status:unavailable', 'service:b'])
+            self.assertMetric('haproxy.count_per_status', value=1, tags=['status:available', 'service:c'])
+            self.assertMetric('haproxy.count_per_status', value=1, tags=['status:unavailable', 'service:c'])
+        else:
+            self.assertMetric('haproxy.count_per_status', value=6, tags=['status:available'])
+            self.assertMetric('haproxy.count_per_status', value=3, tags=['status:unavailable'])
+
+    @mock.patch('requests.get', return_value=MockResponse())
```

Comment:
You can simply use `Mock` from the `mock` module instead of `MockResponse`.
```
@mock.patch('requests.get', return_value=Mock(content=MOCK_DATA))
```